### ISSUE TITLE
Regression: Empty list [] as input to a nested list of objects is interpreted as null

### ISSUE THREAD

**Issue by danowensdev** [gruntwork-io/terragrunt#3312]
## Describe the bug

Since the latest update to terragrunt (0.64.5), an input of an empty list to a nested list of objects is treated as a null value.

## Steps To Reproduce
Minimal example:
```hcl
// terragrunt.hcl
terraform {
    source = "./module"
}
inputs = {
    myvar = [{
        items = []
    }]
}
```

```hcl
// module/main.tf
variable "myvar" {
  type = list(object({
    items = list(object({
      firstname = string
      surname   = string
    }))
  }))
}

output "myvar" {
  value = [
    for entry in var.myvar : {
      items = [for item in entry.items : {
        item = item
      }]
    }
  ]
}
```
## Expected behavior
With terragrunt 0.64.4, `terragrunt plan` produces a successful plan:
```
Changes to Outputs:
  + myvar = [
      + {
          + items = []
        },
    ]

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.
```
With terragrunt 0.64.5, `terragrunt plan` produces the following:
```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Iteration over null value
│ 
│   on main.tf line 16, in output "myvar":
│   16:       items = [for item in entry.items : {
│   17:         item = item
│   18:       }]
│     ├────────────────
│     │ entry.items is null
│ 
│ A null value cannot be used as the collection in a 'for' expression.
```

## Versions

- Terragrunt version: 0.64.5
- Terraform version: v1.8.1
- Environment details: Mac, also occurs on Linux

**Comment by mhulscher** [comment]
Similarly, when explicitly passing an empty map to a variable, it is actually passed `null`.

```
│ Error: Invalid function argument
│ 
│   on mimir.tf line 645, in resource "helm_release" "mimir":
│  645: %{if length(keys(var.pod_annotations)) != 0}
│     ├────────────────
│     │ while calling keys(inputMap)
│     │ var.pod_annotations is null
│ 
│ Invalid value for "inputMap" parameter: argument must not be null.
```

```hcl
# terragrunt.hcl

inputs = {
  pod_annotations = {}
}
```

```hcl
# variables.tf

variable "pod_annotations" {
  type    = map(string)
  default = {}
}
```

**Comment by levkohimins** [comment]
Thanks guys for your feedback, we are working on fixing it.

**Comment by levkohimins** [comment]
Resolved in [v0.66.1](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.66.1) release.
